controller/system_backup_controller.go

```diff
@@ -683,6 +700,151 @@ func (c *SystemBackupController) GenerateSystemBackup(systemBackup *longhorn.Sys
 	}
 }
 
+func (c *SystemBackupController) BackupVolumes(systemBackup *longhorn.SystemBackup) (map[string]*longhorn.Backup, error) {
+	switch systemBackup.Spec.VolumeBackupPolicy {
+	case longhorn.SystemBackupCreateVolumeBackupPolicyIfNotPresent:
+		return c.backupVolumesIfNotPresent(systemBackup)
+	case longhorn.SystemBackupCreateVolumeBackupPolicyAlways:
+		return c.backupVolumesAlways(systemBackup)
+	}
+	return nil, nil
+}
+
+func (c *SystemBackupController) backupVolumesAlways(systemBackup *longhorn.SystemBackup) (map[string]*longhorn.Backup, error) {
+	volumes, err := c.ds.ListVolumesRO()
+	if err != nil {
+		return nil, err
+	}
+
+	volumeBackups := make(map[string]*longhorn.Backup, len(volumes))
+	for _, volume := range volumes {
+		backup, err := c.createVolumeBackup(volume, systemBackup)
+		if err != nil {
+			return nil, err
+		}
+
+		volumeBackups[backup.Name] = backup
+	}
+
+	return volumeBackups, nil
+}
+
+func (c *SystemBackupController) backupVolumesIfNotPresent(systemBackup *longhorn.SystemBackup) (map[string]*longhorn.Backup, error) {
+	volumes, err := c.ds.ListVolumesRO()
+	if err != nil {
+		return nil, err
+	}
+
+	volumeBackups := make(map[string]*longhorn.Backup, len(volumes))
+	for _, volume := range volumes {
+		if volume.Status.LastBackup != "" {
```

Comment:
Will this field be unset when the last backup is removed?

Reply:
https://github.com/longhorn/longhorn-manager/blob/023dd3cbd5009f390608ea9cdd9f31bd397cf8a2/controller/volume_controller.go#L4270-L4271?